File: examples/JvScreenCapture/BCB/JvScreenCaptureMainFormU.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "JvScreenCaptureMainFormU.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "JclGraphics"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button1Click(TObject *Sender)
{
  Graphics::TBitmap *B;

   B = new Graphics::TBitmap();
   try
   {
     ScreenShot(B, true);  // Default value does not exist in C5
     this->Image1->Picture->Bitmap->Assign(B);
   }
   __finally
   {
     delete B;
   }

}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button2Click(TObject *Sender)
{

  Word AFormat;
  THandle AData;
  HPALETTE APalette;
  TClipboard *Clipboard;

   this->Image1->Picture->Bitmap->SaveToClipboardFormat(AFormat, AData, APalette);

   Clipboard = new TClipboard();
   try
   {
    Clipboard->SetAsHandle(AFormat, AData);
   }
   __finally
   {
     delete Clipboard;
   }
}
//---------------------------------------------------------------------------

File: examples/JvScreenCapture/JvScreenCaptureMainFormU.pas
unit JvScreenCaptureMainFormU;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  ExtCtrls, StdCtrls, clipbrd;

type
  TJvScreenCaptureMainForm = class(TForm)
    Button1: TButton;
    Image1: TImage;
    Button2: TButton;
    procedure Button1Click(Sender: TObject);
    procedure Button2Click(Sender: TObject);
  end;

var
  JvScreenCaptureMainForm: TJvScreenCaptureMainForm;

implementation

{$R *.DFM}

uses
  JvFunctions;

procedure TJvScreenCaptureMainForm.Button1Click(Sender: TObject);
var
  b : TBitmap;
begin
   b := CaptureScreen;
   self.image1.picture.bitmap.assign(b);
   b.Free;
end;

procedure TJvScreenCaptureMainForm.Button2Click(Sender: TObject);
var
  MyFormat : Word;
  AData: THandle;
  APalette:HPalette;
begin
   self.image1.picture.bitmap.SaveToClipboardFormat(MyFormat,AData,APalette);
   ClipBoard.SetAsHandle(MyFormat,AData);
end;

end.
